compiler/rustc_pattern_analysis/src/constructor.rs

```diff
@@ -979,7 +979,7 @@ impl ConstructorSet {
             && !(pcx.is_top_level && matches!(self, Self::NoConstructors))
         {
             // Treat all missing constructors as nonempty.
```

Comment:
```suggestion
            // Treat all missing constructors as nonempty.
            // This clears `missing_empty`.
```
I find it nonobvious that this now clears `missing_empty`